nixos/modules/services/networking/nbd.nix

```
@@ -0,0 +1,152 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nbd;
+  configFormat = pkgs.formats.ini { };
+  iniFields = with types; attrsOf (oneOf [ bool int float str ]);
+  serverConfig = configFormat.generate "nbd-server-config"
+    ({
+      generic = ({
+        user = "root";
+        group = "root";
+        port = cfg.server.listenPort;
+      }
+      // cfg.server.extraOptions
+      // (if (cfg.server.listenAddress != null)
+      then { listenaddr = cfg.server.listenAddress; }
+      else { }));
+    }
+    // (mapAttrs
+      (_: { path, allowAddresses, extraOptions }: {
+        exportname = path;
+      }
+      // extraOptions
+      // (if (allowAddresses != null)
+      then {
+        authfile = pkgs.writeText "authfile" ''
+          ${concatStringsSep "\n" allowAddresses}
+        '';
```

```suggestion
        authfile = pkgs.writeText "authfile" (concatStringsSep "\n" allowAddresses);
```
unless the trailing newline added by the interpolation is vital